src/qt/bitcoingui.cpp

```diff
@@ -1084,6 +1091,26 @@ void BitcoinGUI::setEncryptionStatus(int status)
 }
 #endif // ENABLE_WALLET
 
+void BitcoinGUI::updateProxyIcon()
+{
+    bool proxy_enabled = false;
+    proxyType proxy;
+
+    for (int n = 0; n < NET_MAX; ++n) {
```

Comment:
`GetProxy` returns `IsValid` so something like this should work:
```cpp
for (int n = 0; !proxy_enabled && n < NET_MAX; ++n) {
    proxy_enabled = GetProxy((Network) n, proxy);
}
```

Reply:
Nice, I didn't notice GetProxy checks for valid.